documentation/functions/domain/IGNORE.md

```diff
@@ -1,8 +1,162 @@
+
+typeSpec and targetSpec default to `"*"` if they are omitted.
+
+## Globs
+
+The `labelSpec` and `targetSpec` parameters supports glob patterns in the style
+of the [gobwas/glob](https://github.com/gobwas/glob) library.  All of the
+following patterns will work:
+
+* `IGNORE("*.foo")` will ignore all records in the style of `bar.foo`, but will not ignore records using a double
+subdomain, such as `foo.bar.foo`.
+* `IGNORE("**.foo")` will ignore all subdomains of `foo`, including double subdomains.
+* `IGNORE("?oo")` will ignore all records of three symbols ending in `oo`, for example `foo` and `zoo`. It will
+not match `.`
+* `IGNORE("[abc]oo")` will ignore records `aoo`, `boo` and `coo`. `IGNORE("[a-c]oo")` is equivalent.
+* `IGNORE("[!abc]oo")` will ignore all three symbol records ending in `oo`, except for `aoo`, `boo`, `coo`.        `IGNORE("[!a-c]oo")` is equivalent.
+* `IGNORE("{bar,[fz]oo}")` will ignore `bar`, `foo` and `zoo`.
+* `IGNORE("\\*.foo")` will ignore the literal record `*.foo`.
+
+## Examples
+
+General examples:
+
+{% code title="dnsconfig.js" %}
+```javascript
+D("example.com",
+  IGNORE("foo"), // matches any records on foo.example.com
+  IGNORE("baz", "A"), // matches any A records on label baz.example.com
+  IGNORE("*", "MX", "*"), // matches all MX records
+  IGNORE("*", "CNAME", "dev-*"), // matches CNAMEs pointing to hosts that start with dev-*
+  IGNORE("bar", "A,MX"), // ignore only A and MX records for name bar
+  IGNORE("*", "*", "dev-*), // Ignore targets with a `dev-` prefix
+  IGNORE("*", "A", "1\.2\.3\."), // Ignore targets that match the 1.2.3.0/24 CIDR block.
+);
+```
+{% endcode %}
+
+Ignore Let's Encrypt (ACME) validation records:
+
+```
+  IGNORE("_acme-challenge.**", "TXT"),
+```
+
+Ignore DNS records typically inserted by Microsoft ActiveDirectory:
+
+```
+  IGNORE("_gc.**", "SRV"), // General Catalog
+  IGNORE("_kerberos.**", "SRV"), // Kerb5 server
+  IGNORE("_kpasswd.**", "SRV"), // Kpassword
+  IGNORE("_ldap.**", "SRV"), // LDAP
+  IGNORE("_msdcs", "NS"), // Microsoft Domain Controller Service
+  IGNORE("_vlmcs.**", "SRV"), // FQDN of the KMS host
+  IGNORE("domaindnszones", "A"),
+  IGNORE("forestdnszones", "A"),
+```
+
+## Don't insert and ignore the same item
+
+It is considered as an error to try to ignore records that
+you yourself are inserting into a domain.
+
+This will generate an error:
+
+```
+D("example.com", ...
+    ...
+    TXT("myhost", "mytext"),
+    IGNORE("myhost", "*", "*"),
+    ...
+```
+
+To disable this safety check, add the `DISABLE_IGNORE_SAFETY_CHECK` statement to the `D()`.
+
+```
+D("example.com", ...
+    DISABLE_IGNORE_SAFETY_CHECK,
+    ...
+    TXT("myhost", "mytext"),
+    IGNORE("myhost", "*", "*"),
+    ...
+```
+
+FYI: Previously DNSControl permitted disabling this check on
+a per-record basis using `IGNORE_NAME_DISABLE_SAFETY_CHECK`:
+
+```
+    // THIS NO LONGER WORKS! Use DISABLE_IGNORE_SAFETY_CHECK instead.
```

Comment:
This is an great one to add on the  changelog breaking chances section. This allows me, as a user, to determine the impact early on. Otherwise I update DNSControl and cannot run the commands before having to change the DNSControl domain configuration. 

Having said that, what version number would you like to make this change under?

Reply:
it starts with diff2.  so... v4.0.0 with "--diff2=false" as the temporary work-around?

Reply:
Yes, that could be a solution.